File: integration/tools/send_payment.py
"""send_payment: compose, sign, and broadcast a Bitcoin payment."""

from __future__ import annotations

from communication.channel.agent_channel import AgentChannel
from integration.rpc.methods import RpcMethod
from integration.rpc.schema import SendPaymentParams, SendPaymentResult


class SendPaymentMethod(RpcMethod):
    """JSON-RPC method that calls AgentChannel.compose_payment + broadcast_payment."""

    name = "send_payment"
    params_schema = SendPaymentParams
    result_schema = SendPaymentResult

    async def __call__(
        self,
        channel: AgentChannel,
        params: SendPaymentParams,
    ) -> SendPaymentResult:
        # Compose BTCPayload, broadcast via the configured Broadcaster, return Txid.
        ...

File: skills/openclaw-trustroom/src/tools/send_payment.ts
// send_payment: compose, sign, and broadcast a Bitcoin payment to a peer.

import type { JSONRpcClient } from "../client.js";

export interface SendPaymentParams {
    recipient: { pubkey_hex: string };
    amount_sats: number;
}

export interface SendPaymentResult {
    txid: string;
}

export const sendPayment = {
    name: "send_payment",
    description: "Compose, sign, and broadcast a Bitcoin payment to another agent. Returns the on-network transaction id.",
    handler: async (
        params: SendPaymentParams,
        ctx: { client: JSONRpcClient },
    ): Promise<SendPaymentResult> => {
        // Forward to the Python sidecar via JSONRpcClient.call("send_payment", params).
        return ctx.client.call<SendPaymentResult>("send_payment", params);
    },
};
